File: app/src/main/java/com/avoid/playtolearn/activities/SettingsActivity.java
package com.avoid.playtolearn.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.avoid.playtolearn.R;
import com.avoid.playtolearn.common.Session;
import com.avoid.playtolearn.widgets.Switch;

public class SettingsActivity extends AppCompatActivity {
    private Switch soundsSwitch;
    private Switch musicSwitch;
    private Switch hintsSwitch;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        this.soundsSwitch = findViewById(R.id.sound_switch);
        this.musicSwitch = findViewById(R.id.music_switch);
        this.hintsSwitch = findViewById(R.id.hints_switch);

        refreshUI();
    }

    public void closeButtonOnClick(View view) {
        finish();
    }

    public void resetButtonOnClick(View view) {
        Session.currentSettings.setSoundsOn(Session.currentSettings.isSoundsOnDefault());
        Session.currentSettings.setMusicOn(Session.currentSettings.isMusicOnDefault());
        Session.currentSettings.setHintsOn(Session.currentSettings.isHintsOnDefault());
        Session.settingsHelper.saveSettings();

        applySoundSettings(SettingsActivity.this);
        applyMusicSettings(SettingsActivity.this);

        refreshUI();
    }

    public void okButtonOnClick(View view) {
        Session.settingsHelper.saveSettings();
        finish();
    }

    public void soundsSwitchOnClick(View view) {
        Session.currentSettings.setSoundsOn(this.soundsSwitch.isEnabled());
        applySoundSettings(SettingsActivity.this);
    }

    public void musicSwitchOnClick(View view) {
        Session.currentSettings.setMusicOn(this.musicSwitch.isEnabled());
        applyMusicSettings(SettingsActivity.this);
    }

    public void hintsSwitchOnClick(View view) {
        Session.currentSettings.setHintsOn(this.hintsSwitch.isEnabled());
    }

    public void refreshUI(){
        this.soundsSwitch.setChecked(Session.currentSettings.isSoundsOn());
        this.musicSwitch.setChecked(Session.currentSettings.isMusicOn());
        this.hintsSwitch.setChecked(Session.currentSettings.isHintsOn());
    }


    private void applyMusicSettings(SettingsActivity settingsActivity) {

    }

    private void applySoundSettings(SettingsActivity settingsActivity) {

    }


}

File: app/src/main/java/com/ivantha/playtolearn/activity/SettingsActivity.kt
package com.ivantha.playtolearn.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.ivantha.playtolearn.R
import com.ivantha.playtolearn.common.SettingsHelper
import kotlinx.android.synthetic.main.activity_settings.*

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        soundSwitch.setOnClickListener({
            SettingsHelper.currentSettings!!.soundsOn = soundSwitch.isChecked
            applySoundSettings()
        })

        musicSwitch.setOnClickListener({
            SettingsHelper.currentSettings!!.musicOn = musicSwitch.isChecked
            applyMusicSettings()
        })

        hintsSwitch.setOnClickListener({
            SettingsHelper.currentSettings!!.hintsOn = hintsSwitch.isChecked
        })

        closeButton.setOnClickListener({
            finish()
        })

        resetButton.setOnClickListener({
            SettingsHelper.currentSettings!!.soundsOn = SettingsHelper.currentSettings!!.soundsOnDefault
            SettingsHelper.currentSettings!!.musicOn = SettingsHelper.currentSettings!!.musicOnDefault
            SettingsHelper.currentSettings!!.hintsOn = SettingsHelper.currentSettings!!.hintsOnDefault
            SettingsHelper.saveSettings(applicationContext)

            applySoundSettings()
            applyMusicSettings()

            refreshUI()
        })

        okButton.setOnClickListener({
            SettingsHelper.saveSettings(applicationContext)
            finish()
        })

        refreshUI()
    }

    private fun refreshUI() {
        soundSwitch.isChecked = SettingsHelper.currentSettings!!.soundsOn
        musicSwitch.isChecked = SettingsHelper.currentSettings!!.musicOn
        hintsSwitch.isChecked = SettingsHelper.currentSettings!!.hintsOn
    }

    private fun applyMusicSettings() {

    }

    private fun applySoundSettings() {

    }

}
